app/models/job_template.rb

```diff
@@ -180,7 +197,6 @@ def sync_foreign_input_sets(input_sets)
   def sync_feature(feature_name)
     if feature_name && (feature = RemoteExecutionFeature.feature(feature_name))
       feature.job_template ||= self
-      feature.save!
```

Comment:
Yes, the `save!` should've never been here, it causes a cascade save forcing the template to be saved.